### PR TITLE
enhance: Load raw data while scalar index doesn't have raw data

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #28886

### CODE REVIEW

**Comment by xiaocai2333** [comment]
run-cpu-e2e

File: internal/core/src/query/visitors/ExecExprVisitor.cpp [hunk 1]
```diff
@@ -2297,8 +2305,16 @@ ExecExprVisitor::ExecCompareExprDispatcher(CompareExpr& expr, Op op)
...
                             field_id, chunk_id);
-                        return [&indexing](int i) -> const number {
-                            return indexing.Reverse_Lookup(i);
+                        if (indexing.HasRawData()) {
+                            return [&indexing](int i) -> const number {
+                                return indexing.Reverse_Lookup(i);
+                            };
+                        }
+                        auto chunk_data =
+                            segment_.chunk_data<bool>(field_id, chunk_id)
```
Comment (yah01) [inline internal/core/src/query/visitors/ExecExprVisitor.cpp:2314]: the type parameter should be `int8_t`, and all the code below should be changed

File: internal/core/src/segcore/SegmentSealedImpl.cpp [hunk 2]
```diff
@@ -1020,33 +1020,14 @@ SegmentSealedImpl::fill_with_empty(FieldId field_id, int64_t count) const {
     return CreateScalarDataArray(count, field_meta);
 }
 
+// DO NOT directly access the column byh map like: `fields_.at(field_id)->Data()`,
```
Comment (yah01) [inline internal/core/src/segcore/SegmentSealedImpl.cpp:1023]: typo `byh` -> `by`

File: internal/core/src/segcore/SegmentSealedImpl.cpp [hunk 3]
```diff
@@ -1070,33 +1070,14 @@ SegmentSealedImpl::fill_with_empty(FieldId field_id, int64_t count) const {
     return CreateScalarDataArray(count, field_meta);
 }
 
+// DO NOT directly access the column by map like: `fields_.at(field_id)->Data()`,
```
Comment (czs007) [inline internal/core/src/segcore/SegmentSealedImpl.cpp:1073]: wrong comment?

**Comment by xiaocai2333** [comment]
rerun ut

**Pull request merged**